srsenb/src/upper/rrc.cc

```diff
@@ -1881,6 +1901,113 @@ void rrc::ue::send_ue_cap_enquiry()
 
 /********************** HELPERS ***************************/
 
+bool rrc::ue::select_security_algorithms() {
+  // Each position in the bitmap represents an encryption algorithm:
+  // “all bits equal to 0” – UE supports no other algorithm than EEA0,
+  // “first bit” – 128-EEA1,
+  // “second bit” – 128-EEA2,
+  // “third bit” – 128-EEA3,
+  // other bits reserved for future use. Value ‘1’ indicates support and value
+  // ‘0’ indicates no support of the algorithm.
+  // Algorithms are defined in TS 33.401 [15].
+  // Note: information missing
+
+  bool enc_algo_found = false;
+  bool integ_algo_found = false;
+  bool zero_vector = true;
+  int i = 0;
+  for (i = 0; i < srslte::CIPHERING_ALGORITHM_ID_N_ITEMS; i++) {
+    switch (parent->cfg.eea_preference_list[i]) {
+    case srslte::CIPHERING_ALGORITHM_ID_EEA0:
+      // “all bits equal to 0” – UE supports no other algorithm than EEA0,
+      zero_vector = true;
+      for (int j; j < LIBLTE_S1AP_ENCRYPTIONALGORITHMS_BIT_STRING_LEN; j++) {
+        if (security_capabilities.encryptionAlgorithms.buffer[j]) {
```

Comment:
why is j not initialized in the for loop?

Reply:
If I do set it to zero, there is something going wrong later on and the traffic isn't working

Reply:
OK, so the only reason I can think that not initializing the loop can break stuff is that the uninitialised `j` is always larger than `LIBLTE_S1AP_ENCRYPTIONALGORITHMS_BIT_STRING_LEN`, thus `zero_vector` == true even if it UE supports different algos, thus the eNB will pick EEA0. Not sure why this will break stuff later on though...